airflow/jobs/scheduler_job.py

```diff
@@ -1585,15 +1586,16 @@ def _create_dag_runs(self, dag_models: Iterable[DagModel], session: Session) ->
         # as DagModel.dag_id and DagModel.next_dagrun
         # This list is used to verify if the DagRun already exist so that we don't attempt to create
         # duplicate dag runs
-        active_dagruns = (
-            session.query(DagRun.dag_id, DagRun.execution_date)
-            .filter(
-                tuple_(DagRun.dag_id, DagRun.execution_date).in_(
-                    [(dm.dag_id, dm.next_dagrun) for dm in dag_models]
-                )
+
+        filter_dms = [
+            and_(
+                DagRun.dag_id == dm.dag_id,
+                DagRun.execution_date == dm.next_dagrun,
```

Comment:
This is a possible performance regression -- if MSSQL doesn't support this when we might need to look at having a query method specialisation per DB backend

Reply:
done, restricted the change to mssql